sklearn/neighbors/kde.py

```diff
@@ -124,6 +130,12 @@ def fit(self, X, y=None, sample_weight=None):
         sample_weight : array_like, shape (n_samples,), optional
             List of sample weights attached to the data X.
         """
+
+        if self.metric == 'default' and self.p != 2:
+            warnings.warn("The default value of p will change to p=2.",
```

Comment:
p did not exist previously here, unless I'm much mistaken, so this is not applicable.